src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs

```diff
@@ -52,9 +52,15 @@ public static void AddToMap(BoundNode root, Dictionary<SyntaxNode, OneOrMany<Bou
                 var builder = new NodeMapBuilder(additionMap, tree, node);
                 builder.Visit(root);
 
+#if NETCOREAPP
+                // Ensure map is large enough to hold found nodes.
+                map.EnsureCapacity(map.Count + additionMap.Keys.Count);
+#endif
+
                 foreach (CSharpSyntaxNode key in additionMap.Keys)
                 {
-                    if (map.ContainsKey(key))
+                    var added = additionMap.GetAsOneOrMany(key);
```

Comment:
>added [](http://example.com/codeflow?start=24&length=5)

The name feels confusing. The thing is not added yet. Consider renaming. Perhaps "nodesToAdd" ?